bundles/org.openhab.binding.evcc/src/main/java/org/openhab/binding/evcc/internal/EvccHandler.java

```diff
@@ -0,0 +1,436 @@
+import org.openhab.binding.evcc.internal.api.dto.Result;
+import org.openhab.core.library.CoreItemFactory;
+import org.openhab.core.library.types.DateTimeType;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.library.unit.MetricPrefix;
+import org.openhab.core.library.unit.SIUnits;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.thing.Channel;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.thing.binding.builder.ChannelBuilder;
+import org.openhab.core.thing.binding.builder.ThingBuilder;
+import org.openhab.core.thing.type.ChannelTypeUID;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link EvccHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Florian Hotze - Initial contribution
+ */
+@NonNullByDefault
+public class EvccHandler extends BaseThingHandler {
+    private final Logger logger = LoggerFactory.getLogger(EvccHandler.class);
+    private @Nullable EvccAPI evccAPI;
+    private @Nullable ScheduledFuture<?> statePollingJob;
+
+    private @Nullable Result result;
+
+    private boolean batteryConfigured = false;
+    private boolean gridConfigured = false;
+    private boolean pvConfigured = false;
+
+    private int targetSoC = 100;
+    private boolean targetTimeEnabled = false;
+    private ZonedDateTime targetTimeZDT = ZonedDateTime.now();
+
+    public EvccHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command.equals(RefreshType.REFRESH)) {
+            refresh();
+        } else {
+            logger.debug("Handling command {} for channel {}", command, channelUID);
+            String groupId = channelUID.getGroupId();
+            if (groupId == null) {
+                return;
+            }
+            String channelIdWithoutGroup = channelUID.getIdWithoutGroup();
+            int loadpoint = Integer.parseInt(groupId.toString().substring(9));
+            EvccAPI evccAPI = this.evccAPI;
+            if (evccAPI == null) {
+                return;
+            }
+            try {
+                switch (channelIdWithoutGroup) {
+                    case CHANNEL_LOADPOINT_MODE:
+                        evccAPI.setMode(loadpoint, command.toString());
+                        break;
+                    case CHANNEL_LOADPOINT_MIN_SOC:
+                        evccAPI.setMinSoC(loadpoint, Integer.parseInt(command.toString().replaceAll(" %", "")));
```

Comment:
Will this work?
```suggestion
                        if (command instanceof PercentType) {
                            evccAPI.setMinSoC(loadpoint, ((PercentType) command).intValue());
                        }
```

Reply:
Yes, in fact that works, but your approach looks much better.

Reply:
How would that work for (current is QuantityType):

```java
 case CHANNEL_LOADPOINT_MIN_CURRENT:
   evccAPI.setMinCurrent(loadpoint, Integer.parseInt(command.toString().replaceAll(" A", "")));
   break;
```

Reply:
I will try do handle the command that way for all channels.

Reply:
> Yes, in fact that works, but your approach looks much better.

Sorry, I meant to ask if my proposal would work. :)

Reply:
Your proposal works slightly modified, thanks.
I have adjusted all command channels accordingly.

See 978656d

Reply:
Very good.